### PR TITLE
Revert "Add environment variable (COMPlus_EnableDiagnostics) to disab…

### PR DESCRIPTION
**Pull request by RussKeldorph**
…le debugging and profiling. (#15878)"

This reverts commit 5bcfde404803f85451cf0ee9fd6406734cb878ff.

If the armel CI jobs don't hang on this, then that suggests #15878 is responsible for #15914.

@mikem8361 FYI

### CODE REVIEW

File: src/debug/ee/debugger.cpp [hunk 1]
```diff
@@ -2085,6 +2103,22 @@ HRESULT Debugger::Startup(void)
 
         RaiseStartupNotification();
 
+        // Also initialize the AppDomainEnumerationIPCBlock
```
Comment (jkotas) [inline src/debug/ee/debugger.cpp]: @mikem8361 Notice that `FEATURE_DBGIPC_TRANSPORT_VM` block above uses `m_pAppDomainCB` - that is initialized too late now.

**Comment by benaadams** [comment]
Sort of worked; but not reporting success to CI
```
Perform an action if the job was performed on an Azure VM Agent. is waiting for a checkpoint on 
dotnet_coreclr » master » armel_cross_checked_tizen_prtest #2999
[Current build status] check if current [SUCCESS] is worse or equals then [SUCCESS] and better or equals then [SUCCESS]
Run condition [Current build status] enabling perform for step [[Archive the artifacts]]
Archiving artifacts
Notifying endpoint with credentials id 'helix-int-notification-url'
Notifying endpoint with credentials id 'helix-prod-notification-url'
Setting status of 8e42b4cdba2607956477336d6bc3ac2aa0cd57c4 to SUCCESS with url https://ci.dot.net/job/dotnet_coreclr/job/master/job/armel_cross_checked_tizen_prtest/3000/ and message: 'Build finished. '
Using context: Tizen armel Cross Checked Build
[WS-CLEANUP] Deleting project workspace...Cannot delete workspace
 :remote file operation failed:
 /mnt/j/workspace/dotnet_coreclr/master/armel_cross_checked_tizen_prtest at hudson.remoting.Channel@6e0ba86f:ubuntu1404-20170120-9d4ad3:
 java.io.IOException: Unable to delete '/mnt/j/workspace/dotnet_coreclr/master/armel_cross_checked_tizen_prtest/Tools/15.0/Microsoft.Common.props'.
 Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
ERROR: Step ‘Delete workspace when build is done’ failed:
 Cannot delete workspace:
 remote file operation failed:
 /mnt/j/workspace/dotnet_coreclr/master/armel_cross_checked_tizen_prtest at hudson.remoting.Channel@6e0ba86f:ubuntu1404-20170120-9d4ad3:
 java.io.IOException: Unable to delete '/mnt/j/workspace/dotnet_coreclr/master/armel_cross_checked_tizen_prtest/Tools/15.0/Microsoft.Common.props'.
 Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
Finished: SUCCESS
```

File: src/debug/ee/debugger.cpp (hunk 1 above)
Comment (mikem8361) [reply src/debug/ee/debugger.cpp]: My change that is being reverted moved the m_pAppDomainCB initialization to almost the beginning of Debugger::Startup() so it should be initialized.

The change is moving the m_AppDomainCB back to where it was (later in Startup()).

Am I missing something?

**Comment by mikem8361** [comment]
The error you hit seems to be a build infrastructure problem.

I really hope we won't have to merge this reversion.  I don't see how this could affect only Linux Arm tests. I don't know how to debug this environment .

**Comment by benaadams** [comment]
CentOS7.1 x64 Release Innerloop Build and Test seems infrastructure 
```
First time build. Skipping changelog.
parallel {
    Schedule job dotnet_coreclr » master » x64_release_centos7.1_innerloop_prtest
    Build dotnet_coreclr » master » x64_release_centos7.1_innerloop_prtest #12 started
    dotnet_coreclr » master » x64_release_centos7.1_innerloop_prtest #12 completed 
ERROR: Failed to run DSL Script
java.util.concurrent.ExecutionException:
dbees.plugins.flow.JobNotFoundException:
net_coreclr/master/x64_release_windows_nt_innerloop_bld_prtest not found (or isn't a job).
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at java_util_concurrent_Future$get.call(Unknown Source)
	at com.cloudbees.plugins.flow.FlowDelegate$_parallel_closure6.doCall(FlowDSL.groovy:456)
	at sun.reflect.GeneratedMethodAccessor1199.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at groovy.lang.Closure.call(Closure.java:414)
	at groovy.lang.Closure.call(Closure.java:430)
```

**Comment by mikem8361** [comment]
Another infrastructure problem.


-------- Original message --------

File: src/debug/ee/debugger.cpp (hunk 1 above)
Comment (jkotas) [reply src/debug/ee/debugger.cpp]: Sorry, I go it backward ... you are right. But I do think that the change of the order here caused the regression.

Comment (jkotas) [reply src/debug/ee/debugger.cpp]: And it is likely that we have the same regression on other Linux flavors and just do not see it as often.

**Comment by jkotas** [comment]
> I don't see how this could affect only Linux Arm tests.

There is a strong evidence that it did affect Linux Arm tests. They started consistently failing right after your change went in; and this trial revert made them pass again.

**Comment by RussKeldorph** [comment]
> I really hope we won't have to merge this reversion. I don't see how this could affect only Linux Arm tests. I don't know how to debug this environment .

@mikem8361 I'm not insisting that merging this is the right fix.  I just wanted to demonstrate that #15878 is the likely culprit.  I think @jashook or @Samsung can help get you set up to debug in this environment if you can't figure it out with @jkotas thoughts.  I would ask that you make it your top priority to investigate this and consider merging this PR if you can't, however.

**Comment by mikem8361** [comment]
I'm on vacation until Monday so you probably want to merge this.


-------- Original message --------

File: src/debug/ee/debugger.cpp [hunk 2]
```diff
@@ -7186,8 +7220,7 @@ void Debugger::JitAttach(Thread * pThread, EXCEPTION_POINTERS * pExceptionInfo,
     }
     CONTRACTL_END;
 
-    // Don't do anything if there is a native debugger already attached or the debugging support has been disabled.
-    if (IsDebuggerPresent() || m_pRCThread == NULL)
```
Comment (benaadams) [inline src/debug/ee/debugger.cpp]: Aside, should this be reversed?
```cpp
if (m_pRCThread == NULL || IsDebuggerPresent())
```

**Comment by mikem8361** [comment]
It shouldn't matter. IsDebuggerPresent a API call to Windows.

**Comment by RussKeldorph** [comment]
CentOS Release failure is #16016.  Merging per above.  We can debug as time allows and make another pull request when ready.   @lt72 FYI

**Pull request merged**